docs/source/design.rst

```diff
@@ -0,0 +1,31 @@
+Design Philosophy
+=================
+
+mirar ia a modular python package for analaysis astronomy images.
+
+There are a few guiding principles that ensures the modularity works:
+
+- Transparent requirements: Processors should either be self-contained, or be explicit about their requirements. Processors have a dedicated function, check_prerequisites, for this purpose. The goal is that processors should work out of the box, regardless of preceding processors, or fail immediately because a required precursor is missing.
+
+- Flexibility for input images: There are a small number of fields which will always be provided in every image. These are core fields. Processors should not assume other keys are present, unless they are generated by explicitly required prerequisites.
+
+- Consistent data: astronomy data might look different when it comes out of the instrument, but when it's read in by mirar the pipeline will make sure that the data is converted to the right format. You should only ever open fits files using the open_fits function in mirar.io, and never use astropy.io directly. The open_fits is a wrapper which also performs additional checks as needed.
+
+- Consistent naming: mirar follows the official python naming conventions, as enumerated in pep:
+    - Classes are CamelCase
+    - Functions are snake_case
+    - Variables are snake_case
+    - Constants are ALL_CAPS
+    - Modules are lowercase
+    - Filenames are lowercase
+    - Directories are lowercase
+
+- Optional image saving: A lot of astronomy software will repeatedly write and rewrite science images. mirar DOES NOT DO THIS! If a processor requires an image to be written, e.g because it is a wrapper to other software, it should be saved as a temporary file then reloaded and the temporary image deleted. The only time an image gets permanently saved is if an ImageSaver processor is used.
```

Comment:
Is this true? We have weight images, scorr images, unc images that get saved by processors.

Reply:
How about now? I reworded it. It does hold for science images and weights, but not others.